### PR TITLE
batch-item code refactoring 

### PR DESCRIPTION
**Pull request by MemorySounds**
Issue #42 

From @haydnba's feedback on one of my previous PRs, I've made updates to make the code a bit tighter.

PR includes:

- Moved the sorting quantity logic to `onSubmit` (only happens once now, as opposed to each time the quantity input-number is incremented) [_Better for performance_]
- Refactored the handling of item creation without images (instead of a dedicated route, we now simply use a query string in the original `createItem()` endpoint) [_cleaner code_]
- Consolidated the various batch-item util functions into one file `batchItemHelpers.js` and added comments to document what they do and why. [_cleaner codebase & better documentation_]

### CODE REVIEW

File: server/services/items.js [hunk 1]
```diff
@@ -754,5 +746,5 @@ module.exports = {
   createBatchItem,
   getBatchItem,
   updateBatchItem,
-  createItemWithoutImageUpload,
+  // createItemWithoutImageUpload,
```
Comment (haydnba) [inline server/services/items.js:749]: can go

File: client/src/components/molecules/EditForm/ItemCreateForm.js [hunk 2]
```diff
@@ -39,10 +39,25 @@ export const ItemCreateForm = (data) => {
...
+    const sizeOrder =
+      formValues.shoeSizes.length > 0 ? shoeSizeOptions : clothingSizeOptions;
+    const quantities = formValues[fieldName];
+    const keyValueArray = Object.entries(quantities);
+    keyValueArray.sort(
+      (a, b) => sizeOrder.indexOf(a[0]) - sizeOrder.indexOf(b[0])
+    );
+    const sortedQuantities = Object.fromEntries(keyValueArray);
+    return { ...formValues, [fieldName]: sortedQuantities };
+  };
```
Comment (haydnba) [inline client/src/components/molecules/EditForm/ItemCreateForm.js:54]: good idea!

File: client/src/utils/batchItemHelpers.js [hunk 3]
```diff
@@ -0,0 +1,149 @@
+import {
+  deleteItem,
+  updateItem,
+  getBatchItem,
+  updateBatchItem,
+} from '../services/items';
+
```
Comment (haydnba) [inline client/src/utils/batchItemHelpers.js:7]: This is really helpful, thanks Robert 👍🏼

File: client/src/pages/Item/Item.js [hunk 4]
```diff
@@ -167,19 +175,28 @@ export const Item = () => {
...
-                    quantity={quantity}
-                    afterAddToBasket={resetSizeAndQuantity}
-                  />
-                ) : (
-                  <AddItemToBasket item={itemDetails} limit={limit} />
-                ))}
+              {isShopped ? (
+                <RemoveFromBasketButton itemId={itemDetails._id}>
+                  Remove from Basket
+                </RemoveFromBasketButton>
```
Comment (haydnba) [inline client/src/pages/Item/Item.js:181]: This seems very sensible

File: client/src/components/molecules/EditForm/ItemCreateForm.js (hunk 2 above)
Comment (haydnba) [reply client/src/components/molecules/EditForm/ItemCreateForm.js:54]: Just being lazy and want to make sure this sorting will be happening not only on the item creation but also on item updates??

**Review by haydnba — APPROVED** [review]
Looks great Robert thanks so much for this 💯

File: client/src/components/molecules/EditForm/ItemCreateForm.js (hunk 2 above)
Comment (MemorySounds) [reply client/src/components/molecules/EditForm/ItemCreateForm.js:54]: A yikes, a very good point that I overlooked. 

I'll push the changes in for now (as they now contain some fixes that I'd like to get re-tested), and then will address this in the next PR :)

**Pull request merged**